src/psyclone/dynamo0p3.py

```diff
@@ -9110,62 +9147,115 @@ def ref_name(self, function_space=None):
                     break
             if not found:
                 raise GenerationError(
-                    "DynKernelArgument:ref_name(fs). The supplied function "
+                    "DynKernelArgument.ref_name(fs): The supplied function "
                     "space (fs='{0}') is not one of the function spaces "
-                    "associated with this argument (fss='{1}')".format(
+                    "associated with this argument (fss={1}).".format(
                         function_space.orig_name,
                         self.function_space_names))
-        if self._type in LFRicArgDescriptor.VALID_FIELD_NAMES:
+        if self.is_field():
             return "vspace"
-        if self.is_operator:
+        if self.is_operator():
             if function_space.orig_name == self.descriptor.function_space_from:
                 return "fs_from"
             elif function_space.orig_name == self.descriptor.function_space_to:
                 return "fs_to"
             else:
                 raise GenerationError(
-                    "ref_name: Error, function space '{0}' is one of the "
-                    "gh_operator function spaces '{1}' but is not being "
-                    "returned by either function_space from '{2}' or "
-                    "function_space_to '{3}'".format(
+                    "DynKernelArgument.ref_name(fs): Function space '{0}' "
+                    "is one of the 'gh_operator' function spaces '{1}' but "
+                    "is not being returned by either function_space from "
+                    "'{2}' or function_space_to '{3}'.".format(
                         function_space.orig_name, self.function_spaces,
                         self.descriptor.function_space_from,
                         self.descriptor.function_space_to))
         else:
             raise GenerationError(
-                "ref_name: Error, unsupported arg type '{0}' found".
-                format(self._type))
+                "DynKernelArgument.ref_name(fs): Found unsupported argument "
+                "type '{0}'.".format(self._type))
+
+    def is_scalar(self):
```

Comment:
I agree with you but I think it would then be good to do this from the base `KernelArgument` where `is_scalar()` is first defined. This means changes there and in `GOcean` classes, code and tests as well. I'm happy to consolidate them in this PR if you're happy with it.

Reply:
P.S. As said in a comment above, I changed all these methods into properties (starting with the abstract property `is_scalar` in the base `KernelArgument` class).